features/mbedtls/targets/TARGET_CRYPTOCELL310/aes_alt.c

```diff
@@ -938,7 +964,7 @@ int mbedtls_aes_crypt_ecb( mbedtls_aes_context *ctx,
 /*
  * AES-CBC buffer encryption/decryption
  */
-static int mbedtls_aes_crypt_sw_cbc( struct sw_aes_context *ctx,
+static int mbedtls_aes_crypt_sw_cbc( mbedtls_aes_context *ctx,
```

Comment:
I disagree but its a matter of taste. I mean, the non _sw_ function is only missing two lines to deal with both software and hardware. That is, you only need to add at the top something like:
```
if( !ctx->is_sw )
    return CC_aes_cipher( ... );

// The original mbed TLS code here
```
I think this is also very readable and helps you remove the additional function...